onnxruntime/core/platform/windows/env.cc

```diff
@@ -179,8 +185,13 @@ class WindowsEnv : public Env {
   }
 
   Status GetFileLength(_In_z_ const ORTCHAR_T* file_path, size_t& length) const override {
+#if WINVER >= _WIN32_WINNT_WIN8
     wil::unique_hfile file_handle{
-        CreateFileW(file_path, GENERIC_READ, FILE_SHARE_READ, NULL, OPEN_EXISTING, FILE_ATTRIBUTE_NORMAL, NULL)};
```

Comment:
>GENERIC_READ [](start = 31, length = 12)

why switch to FILE_READ_ATTRIBUTES?

Reply:
It's only reading the length instead of contents

Reply:
And that's how it was done elsewhere (in the `GetCanonicalPath` function)